### PR TITLE
DDF-6137 Upgrade Tika to 1.24.1

### PR DESCRIPTION
**Pull request by SmithJosh**
#### What does this PR do?
Tika and its dependencies have a few CVEs reported against them. This PR upgrades to the latest Tika version to fix them. Also, upgrades POI and other transitive Tika dependencies to match the version that Tika uses.

This PR depends on https://github.com/codice/thirdparty/pull/86 

#### Who is reviewing it? 
@blen-desta @garrettfreibott @bakejeyner @stustison 

#### Select relevant component teams: 
@codice/io 
@codice/security 

#### How should this be tested?

1. Build https://github.com/codice/thirdparty/pull/86
2. Build this PR branch and run DDF
3. Download this set of assorted file types: https://github.com/codice/ddf/tree/master/catalog/transformer/catalog-transformer-tika-input/src/test/resources. 
4. Test the Tika input transformer by ingesting those files from the Karaf shell.
```
admin@root()> catalog:ingest -t tika "/path/to/test/resources"
```
5. Test the video input transformer
```
admin@root()> catalog:ingest -t video "/path/to/test/resources/testMP4.m4a"
```
6. Test the pdf input transformer
```
admin@root()> catalog:ingest -t pdf "/path/to/test/resources/testPDF.pdf"
```
7. Test the OgcUrlResourceReader by attempting to download a resource from the UI. You'll need this change to install the UI: https://github.com/codice/ddf-ui/pull/254

#### Any background context you want to provide?
This is part of an ongoing effort to reduce the DDF CVE level

#### What are the relevant tickets?
Fixes: #6137
https://codice.atlassian.net/browse/DDF-1986
https://codice.atlassian.net/browse/DDF-3064

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by SmithJosh** [comment]
build now

**Comment by SmithJosh** [comment]
CI needs the new Tika bundle to be released first

File: features/utilities/src/main/feature/feature.xml [hunk 1]
```diff
@@ -155,4 +159,20 @@
...
     </feature>
 
+    <feature name="jackson" version="${project.version}">
+        <bundle>mvn:com.fasterxml.jackson.core/jackson-core/${jackson.version}</bundle>
+        <bundle>mvn:com.fasterxml.jackson.core/jackson-annotations/${jackson.version}</bundle>
+        <bundle>mvn:com.fasterxml.jackson.core/jackson-databind/${jackson.version}</bundle>
+    </feature>
+
+    <feature name="poi" version="${project.version}">
+        <feature dependency="true">apache-commons</feature>
```
Comment (Lambeaux) [inline features/utilities/src/main/feature/feature.xml:169]: ❓ Oddly enough, just declaring the `<feature>` dependency without attributes works most of the time so I've never understood why the feature data model needed these attributes. 

I've seen cases where`dependency="true"` is necessary, and also cases where it doesn't help at all. There's also `prerequisite="true"` which I think creates a new OSGi subsystem but sometimes works when `dependency="true"` does not. Have you explored these subtle differences by chance and know precisely what differs between the different strategies?

Comment (SmithJosh) [reply features/utilities/src/main/feature/feature.xml:169]: I'm not sure. I didn't actually add it here. This feature is just moved from another file.

In my experience I've not seen it make a difference either. I'm curious too if anybody can explain.

Comment (stustison) [reply features/utilities/src/main/feature/feature.xml:169]: I'm not sure it does anything right now. It is just a flag. I believe they have changed behavior in later versions of Karaf to take that flag into account though. I remember seeing a commit or two related to it as I was digging through karaf commits.

**Comment by bakejeyner** [comment]
Hero ✔️

**Comment by SmithJosh** [comment]
build now

**Pull request merged**